### ISSUE TITLE
Camera configuration is very slow

### ISSUE THREAD

**Issue by demonskull** [gphoto/gphoto2#44]
Hi there,

Thank you very much for such a great tool!

I often use the following command to configure my DSLR (Canon 350D):

gphoto2 --set-config-index /main/settings/iso=10 --set-config-index /main/settings/capturesizeclass=0 --set-config-index /main/settings/shootingmode=4 --set-config-index /main/settings/shutterspeed=16 --set-config-index /main/settings/aperture=12 --set-config-index /main/settings/focusmode=3 --set-config-index /main/settings/imageformat=8

The quoted setting values are just examples.
When I run this command from a Raspberry Pi 3 it takes about 35s to run! From my desktop PC (ubuntu) it takes substantially less time because it is a more powerful computer, but it is still quite a slow command. This tells me that the speed issue is not related to the camera (otherwise the command would require approximately the same time to run in either case).

I am surprised that a text based command like this can be so slow. For example, the time to download a RAW picture from the camera, which usually has a size around 8MB, is much much shorter, typically around 1s, also with the RasPi3.

Could this be a problem with the inner workings of gphoto2? How does gphoto2 process all the parameters I'm sending? Is there a better (faster) way of changing a bunch of settings at a time?

Note: currently using version 2.5.9-3 of gphoto2

Thanks in advance!

D

**Comment by msmeissn** [comment]
well, we are talking back and forth with a device, and so we might get unexpected timeouts.

it should be similar quick on rasperry pi than on fat machines.

can you capture a debug trace, add --debug --debug-logfile=debug.log   
the debug.log file will contain timestamps so we can see what took so long?

**Comment by demonskull** [comment]
[gphoto.txt](https://github.com/gphoto/gphoto2/files/546503/gphoto.txt)
Hi Marcus,

Thank you very much for your reply.
I've attached the debug output as suggested. There are several errors reading from the port happening right before the long delays, but I cannot make sense of them. Hopefully you might know what's going on.

Regards

D

**Comment by msmeissn** [comment]
This is a bug we fixed in the 2.5.6 libgphoto2 release ... we always queried "zoom", but on EOS this value is not changable, as all DSLR lenses have manual zoom 

so -> update to 2.5.6 should get rid of the delays

**Comment by demonskull** [comment]
Hi Marcus,
OK, I'll give it a go and report back. If I'm going to update, I might just go for the most up-to-date version. :)

Once again, thank you for your help!

D

**Comment by msmeissn** [comment]
you need to update libgphoto2, the driver functionality is in there.

gphtoo2 is just the commandline frontend

**Comment by demonskull** [comment]
Hi Marcus,

I have compiled libgphoto2 2.5.10, going through the sequence configure, make and make install. After this, gphoto2 continues to pick a previous version (2.5.9 in this case, as I'm currently updating on my PC). (Note: I have only updated libgphoto2, not gphoto2)
I have looked in the INSTALL file, and it redirects me to your online manual, where it says:

(http://www.gphoto.org/doc/manual/configure-make-install.html)
WRITEME :-P (PKG_CONFIG_PATH, ./configure, make, make install, /etc/ld.so.conf, LD_LIBRARY_PATH, PATH, yadda yadda)

Unfortunately I'm probably stuck with the yadda yadda. Could you please give me hand?

Thanks!

D

**Comment by msmeissn** [comment]
if your pc updated to 2.5.9 already thats fine.

for building, 
- make sure you have libusb development headers installed. (libusb... -devel package)

- ./configure --prefix=/usr  --libdir=/usr/lib/arm-linux-gnueabihf  to make the build overwrite the system ones. (guiessing the library directory on your raspi, it will be different on ubuntu pc)

**Comment by demonskull** [comment]
Hi Marcus,

Thanks once more for your reply.

Last night I gave a try to a gphoto2 installer, you might be familiar with it:
https://github.com/gonzalo/gphoto2-updater

It worked nicely, but I will keep your instructions anyway.

Regarding the delay, just as you forecast, the problem has been solved with the update, so...

THANK YOU VERY MUCH!

I haven't timed it, but there's no need, it's fast as expected.

Best regards,

D
